test/unit/Governor.js

```diff
@@ -0,0 +1,558 @@
+                    value: ["0", "0", "0"],
+                    data: [data0, data1, data2]
+                }
+            )
+
+            truffleAssert.eventEmitted(
+                tx,
+                "UpdateCancelled",
+                e => e.update.target[0] == setUint256.address
+                    && e.update.value[0] == "0"
+                    && e.update.data[0] == data0
+                    && e.update.target[1] == setUint256.address
+                    && e.update.value[1] == "0"
+                    && e.update.data[1] == data1
+                    && e.update.target[2] == setUint256.address
+                    && e.update.value[2] == "0"
+                    && e.update.data[2] == data2,
+                "UpdateCancelled event not emitted correctly"
+            )
+        })
+    })
+
+    describe("executeUpdate", () => {
+        it("reverts when the update has not been staged", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    }
+                ),
+                "update is not staged"
+            )
+        })
+
+        it("reverts when delay for the staged update has not expired", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            await governor.stage(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                },
+                "100"
+            )
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    }
+                ),
+                "delay for update not expired"
+            )
+        })
+
+        it("reverts when one of the remote calls in the batch fails", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            await governor.stage(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                },
+                "100"
+            )
+
+            await setUint256.setShouldFail(true)
+
+            await fixture.rpc.wait(101)
+
+            // test forwarded revert reason
+            await truffleAssert.reverts(governor.execute(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                }
+            ), "I should fail")
+        })
+
+        it("executes an update: delete the update and emit an UpdateExecuted event", async () => {
```

Comment:
Should we also test for successful ETH transfer when the "value" in one of the txs included in an update is non-zero?

Reply:
We could but if this was written the wrong way it wouldn't compile, but we could add it for completeness.

Reply:
I think we should add a test for ETH transfers if the contract is going to support ETH transfers as a part of updates.